wa/workloads/pcmark/uiauto/app/src/main/java/com/arm/wa/uiauto/pcmark/UiAutomation.java

```diff
@@ -0,0 +1,138 @@
+/*    Copyright 2014-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arm.wa.uiauto.pcmark;
+
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+import com.arm.wa.uiauto.ActionLogger;
+import android.util.Log;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    private int networkTimeoutSecs = 30;
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(networkTimeoutSecs);
+    public static String TAG = "UXPERF";
+
+    @Before
+    public void initialize(){
+        initialize_instrumentation();
+    }
+
+    @Test
+    public void setup() throws Exception{
+        setScreenOrientation(ScreenOrientation.NATURAL);
+        loadBenchmarks();
+        installBenchmark();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void teardown() throws Exception{
+        unsetScreenOrientation();
+    }
+
+    //Swipe to benchmarks and back to initialise the app correctly
+    private void loadBenchmarks() throws Exception {
+        UiObject title = 
+            mDevice.findObject(new UiSelector().resourceId("com.futuremark.pcmark.android.benchmark:id/actionBarMenuItemText")
```

Comment:
I've just tested the workload with our nexus 10 / chromebook (tablet layout) and this element can't be found. Maybe use the menu bar itself instead? Resource-id: `android:id/action_bar` as this seems to be consistent across the devices although the class name for that element does change.